rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -1341,8 +1341,9 @@ rmw_take_serialized_message(
   rmw_subscription_allocation_t * allocation)
 {
   static_cast<void>(allocation);
+  rmw_message_info_t message_info;
```

Comment:
The `RMW_CHECK_ARGUMENT_FOR_NULL()` check for `message_info` in __rmw_take_serialized()` should probably be moved to `rmw_take_serialized_message_with_info()`, and ` TRACETOOLS_TRACEPOINT(rmw_take, ...)` could check if `message_info` is `nullptr` before trying to use it (like [in `rmw_cyclonedds`](https://github.com/ros2/rmw_cyclonedds/blob/406277084be4352fa48357047ed56be69e4942e5/rmw_cyclonedds_cpp/src/rmw_node.cpp#L3364)). Then `rmw_take_serialized_message()` can keep using `nullptr` for `message_info`.

Reply:
ah this makes more sense, thanks!